### PR TITLE
Improvements and style

### PR DESCRIPTION
**Pull request by rwaffen**
Had some time while traveling, so i reworked the module. Did some reordering, style fixes and added parameters to all the classes, so that you can test them one by one. Added some spec tests, but not for all the classes. Changed the behavior of the big over-all-test. merge the deb and the rhel version into one file.

### CODE REVIEW

**Comment by rwaffen** [comment]
that's not funny... apt-things are broken again :disappointed:

**Comment by rwaffen** [comment]
Request for Comments … @tuxmea @nmesstorff @michael-os

File: manifests/compute_node/config.pp [hunk 1]
```diff
@@ -29,102 +29,78 @@
...
     source => 'puppet:///modules/one/qemu.conf'
-  }
+  } ->
 
-  # Imaginator
   file { '/var/lib/one/.virtinst':
-    ensure => directory,
-    owner  => 'oneadmin',
-    group  => 'oneadmin',
-    mode   => '0755',
```
Comment (tuxmea) [inline manifests/compute_node/config.pp]: Please keep the mode in place.
You might get trouble when umask is set to a more restrictive setting.

File: manifests/compute_node/config.pp [hunk 2]
```diff
@@ -29,102 +29,78 @@
...
+    owner   => 'oneadmin',
+    group   => 'oneadmin',
+    require => File['/var/lib/one'],
+  } ->
 
   file { '/var/lib/one/.libvirt':
-    ensure => directory,
-    owner  => 'oneadmin',
-    group  => 'oneadmin',
-    mode   => '0755',
```
Comment (tuxmea) [inline manifests/compute_node/config.pp]: same here: mode.

File: manifests/install.pp [hunk 3]
```diff
@@ -6,22 +6,14 @@
   $http_proxy = $one::http_proxy,
   $dbus_pkg   = $one::dbus_pkg,
 ){
+  File['/etc/gemrc'] -> Package <| provider == 'gem' |>
 
   file { '/etc/gemrc':
-    ensure  => 'file',
+    ensure  => file,
     content => "---\nhttp_proxy: ${http_proxy}\n",
```
Comment (tuxmea) [inline manifests/install.pp]: How will gem behave if you have empty $http_proxy variable?

File: manifests/oned/oneflow/service.pp [hunk 4]
```diff
@@ -17,22 +17,11 @@
...
     ensure     => running,
-    provider   => 'base',
-    hasstatus  => false,
-    hasrestart => false,
-    start      => '/bin/su -c "/usr/bin/oneflow-server start" oneadmin',
-    stop       => '/bin/su -c "/usr/bin/oneflow-server stop" oneadmin',
     enable     => true,
-    path       => '/usr/bin/',
-    require    => Service['opennebula']
+    hasstatus  => true,
```
Comment (tuxmea) [inline manifests/oned/oneflow/service.pp]: Have you verified the exit code of the status command?

File: manifests/oned/onegate/service.pp [hunk 5]
```diff
@@ -17,21 +17,11 @@
...
   service {'onegate-server':
     ensure     => running,
-    provider   => 'base',
-    hasstatus  => false,
-    hasrestart => false,
-    start      => '/bin/su -c "/usr/bin/onegate-server start" oneadmin',
-    stop       => '/bin/su -c "/usr/bin/onegate-server stop" oneadmin ',
     enable     => true,
-    path       => '/usr/bin/'
+    hasstatus  => true,
```
Comment (tuxmea) [inline manifests/oned/onegate/service.pp]: Have you verified the exit code of this status command?

File: manifests/oned/sunstone/config.pp [hunk 6]
```diff
@@ -23,26 +23,28 @@
...
-    ensure  => 'file',
     owner   => 'root',
-    group   => 'oneadmin',
   }
   file { '/usr/lib/one/sunstone':
-    ensure  => 'directory',
+    ensure  => directory,
+    # not sure if this is necessary
     owner   => 'oneadmin',
+    group   => 'oneadmin',
```
Comment (tuxmea) [inline manifests/oned/sunstone/config.pp]: please set file resource type mode explicitly unless you are absolutely sure that any mode is irrelevant for service functionality.

File: manifests/init.pp [hunk 7]
```diff
@@ -386,10 +386,12 @@
             $libvirtd_source                = $one::params::libvirtd_source,
             $rubygems                       = $one::params::rubygems,
```
Comment (tuxmea) [inline manifests/init.pp]: you are missing the new one_version hiera parameter.

**Comment by rwaffen** [comment]
thanks for the input... i will have a closer look at these things in the next days

File: manifests/install.pp (hunk 3 above)
Comment (rwaffen) [inline manifests/install.pp]: i've tested that. if its empty, it behaves like there is no proxy set

File: manifests/compute_node/config.pp [hunk 8]
```diff
@@ -29,102 +29,78 @@
...
   file { '/var/lib/one/.virtinst':
-    ensure => directory,
-    owner  => 'oneadmin',
-    group  => 'oneadmin',
-    mode   => '0755',
-  }
+    ensure  => directory,
+    owner   => 'oneadmin',
+    group   => 'oneadmin',
+    require => File['/var/lib/one'],
```
Comment (michael-os) [inline manifests/compute_node/config.pp]: What is this require good for? As far as I understood, file and directory resources have automatic dependency if one is contained within the other?

File: manifests/compute_node/config.pp [hunk 9]
```diff
@@ -29,102 +29,78 @@
...
   file { '/var/lib/one/.libvirt':
-    ensure => directory,
-    owner  => 'oneadmin',
-    group  => 'oneadmin',
-    mode   => '0755',
-  }
+    ensure  => directory,
+    owner   => 'oneadmin',
+    group   => 'oneadmin',
+    require => File['/var/lib/one'],
```
Comment (michael-os) [inline manifests/compute_node/config.pp]: Same here

File: manifests/init.pp [hunk 10]
```diff
@@ -412,9 +414,9 @@
     include one::oned::sunstone
   }
   if($oneflow) {
-    class {'one::oned::oneflow': }
+    include one::oned::oneflow
```
Comment (michael-os) [inline manifests/init.pp]: Missing relation / no containment!?

File: manifests/init.pp [hunk 11]
```diff
@@ -412,9 +414,9 @@
...
   }
   if($onegate) {
-    class {'one::oned::onegate': }
+    include one::oned::onegate
```
Comment (michael-os) [inline manifests/init.pp]: See above

File: manifests/init.pp [hunk 12]
```diff
@@ -386,10 +386,12 @@
...
             $rubygems                       = $one::params::rubygems,
             ) inherits one::params {
+  include one::prerequisites
   include one::install
   include one::config
   include one::service
 
+  Class['one::prerequisites']->
   Class['one::install']->
   Class['one::config']->
```
Comment (michael-os) [inline manifests/init.pp]: Should we go for containment and add start.- and end-anchors? See also unordered included classes below

File: manifests/init.pp [hunk 13]
```diff
@@ -412,9 +414,9 @@
     include one::oned::sunstone
```
Comment (michael-os) [inline manifests/init.pp]: Unordered

**Comment by michael-os** [comment]
Overall a very nice improvement to the module, thx for putting effort into this!

**Comment by rwaffen** [comment]
have to rebase this on and will merge than...

**Pull request merged**